services/blockchain-indexer/shared/dataService/business/tokens.js

```diff
@@ -133,46 +124,33 @@ const getTopLiskAddresses = async (params) => {
 	return topLiskAddresses;
 };
 
-const getSupportedTokens = async (params) => {
-	const supportedTokensList = {
+const getTokensSummary = async (params) => {
+	const summary = {
 		data: {
-			supportedTokens: [],
 		},
 		meta: {},
 	};
 
-	const response = await requestConnector('token_getSupportedTokens');
-
-	supportedTokensList.data.supportedTokens = await BluebirdPromise.map(
-		response.tokenIDs,
-		async (tokenID) => {
-			const tokenMetadataResponse = await getTokenMetadataByID(tokenID);
-			const [tokenMetadata = {}] = tokenMetadataResponse.data || [];
-
-			return {
-				tokenID,
-				symbol: tokenMetadata.symbol,
-				name: tokenMetadata.tokenName,
-			};
-		},
-		{ concurrency: response.tokenIDs.length },
-	);
+	const escrowedAmount = await fetchInfoFromConnector('token_getEscrowedAmounts', params.offset, params.limit);
+	const supportedTokens = await fetchInfoFromConnector('token_getSupportedTokens', params.offset, params.limit);
+	const totalSupply = await fetchInfoFromConnector('token_getTotalSupply', params.offset, params.limit);
```

Comment:
Please create proper interfaces within the connector and invoke them. Do not bypass them and keep the code consistent.

Also, please remove the support for pagination on the endpoint. I've updated the design document accordingly.

Reply:
applied.